source/common/http/filter/gzip_filter.cc

```diff
@@ -0,0 +1,159 @@
+#include "common/http/filter/gzip_filter.h"
+
+#include <regex>
+
+#include "common/common/macros.h"
+
+namespace Envoy {
+namespace Http {
+
+static const std::regex& acceptEncodingRegex() {
+  CONSTRUCT_ON_FIRST_USE(std::regex, "(?!.*gzip;\\s*q=0(,|$))(?=(.*gzip)|(^\\*$))",
+                         std::regex::optimize);
+}
+
+GzipFilterConfig::GzipFilterConfig(const envoy::api::v2::filter::http::Gzip& gzip)
+    : compression_level_(compressionLevelEnum(gzip.compression_level())),
+      compression_strategy_(compressionStrategyEnum(gzip.compression_strategy())),
+      content_length_(static_cast<uint64_t>(gzip.content_length().value())),
+      memory_level_(static_cast<uint64_t>(gzip.memory_level().value())),
+      etag_(gzip.disable_on_etag().value()),
+      last_modified_(gzip.disable_on_last_modified().value()) {
+
+  for (const auto& value : gzip.cache_control()) {
+    cache_control_values_.insert(value);
+  }
+
+  for (const auto& value : gzip.content_type()) {
+    content_type_values_.insert(value);
+  }
+}
+
+ZlibCompressionLevelEnum
+GzipFilterConfig::compressionLevelEnum(const GzipV2CompressionLevelEnum& compression_level) {
+  if (compression_level == GzipV2CompressionLevelEnum::Gzip_CompressionLevel_Enum_BEST) {
```

Comment:
I think this can be done with a `switch` statement since what comes from the proto is a true enum. It'll probably be a bit more readable code than the if statements.

Reply:
I've tried switch here, but for some reason didn't work. Let me try it again.